frontend/src/components/common/UserNameTooltip.tsx

```diff
@@ -0,0 +1,48 @@
+import { useApiUsers } from 'hooks/pims-api/useApiUsers';
+import * as React from 'react';
+import styled from 'styled-components';
+
+import TooltipIcon from './TooltipIcon';
+
+export interface IUserNameTooltipProps {
+  userGuid?: string;
+  userName?: string;
+}
+
+/** Generic user info tooltip component that displays user name on hover */
+export const UserNameTooltip: React.FunctionComponent<IUserNameTooltipProps> = ({
+  userGuid,
+  userName,
+}) => {
+  const { getUserInfo } = useApiUsers();
+  const [userNameInfo, setUserNameInfo] = React.useState<string>('');
+  React.useEffect(() => {
+    if (userGuid) {
+      getUserInfo(userGuid).then(({ data }) => {
```

Comment:
this is fine for this sprint, but in future sprints lets start using the apiWrapper. That handles errors, as well as not updating state when the component is unmounted.

Reply:
For this sprint at least you'll want to add an ismounted check in addition to the check for data.

Reply:
For now I have used ismounted and pushed the changes